// better-auth/dist/index.mjs
export { b as betterAuth, c as createTelemetry, g as getTelemetryAuthConfig } from './shared/better-auth.Cnh5fEd0.mjs';
export { B as BetterAuthError, M as MissingDependencyError } from './shared/better-auth.DdzSJf-n.mjs';
export { c as capitalizeFirstLetter } from './shared/better-auth.D-2CmEwz.mjs';
export { H as HIDE_METADATA, g as generateState, p as parseState } from './shared/better-auth.Dgq_1hxR.mjs';
export { g as generateId } from './shared/better-auth.BUPPRXfK.mjs';
export { c as createLogger, l as levels, a as logger, s as shouldPublishLog } from './shared/better-auth.C4g12FAA.mjs';
export { APIError } from 'better-call';
import './shared/better-auth.CBY7cUGy.mjs';
import 'zod';
import './shared/better-auth.CiuwFiHM.mjs';
import '@better-auth/utils/base64';
import '@better-auth/utils/hmac';
import '@better-auth/utils/binary';
import './shared/better-auth.Dlx1jCB2.mjs';
import './api/index.mjs';
import './shared/better-auth.fV_ra52g.mjs';
import './shared/better-auth.CW6D9eSx.mjs';
import './shared/better-auth.CvNZNAOW.mjs';
import './shared/better-auth.DxV4YGX3.mjs';
import './shared/better-auth.Cwj5CPkV.mjs';
import 'defu';
import './crypto/index.mjs';
import '@better-auth/utils/hash';
import '@noble/ciphers/chacha.js';
import '@noble/ciphers/utils.js';
import 'jose';
import '@noble/hashes/scrypt.js';
import '@better-auth/utils/hex';
import '@noble/hashes/utils.js';
import './shared/better-auth.B4Qoxdgc.mjs';
import '@better-auth/utils/random';
import './shared/better-auth.C_kPikD6.mjs';
import './shared/better-auth.Cjw1GDsh.mjs';
import './shared/better-auth.r7DxcRaY.mjs';
import 'kysely';
import './shared/better-auth.s0y6HeiD.mjs';
import './shared/better-auth.BRPXwKUM.mjs';
import './shared/better-auth.YwDQhoPc.mjs';
import '@better-fetch/fetch';
import 'jose/errors';
